tools/cache-zulip-git-version

```diff
@@ -1,5 +1,12 @@
 #!/usr/bin/env bash
-set -e
+set -eu
 
 cd "$(dirname "$0")/.."
-git describe --tags --match='[0-9]*' >zulip-git-version || true
+{
+    git describe --always --tags --match='[0-9]*'
+    branches="$(git for-each-ref --format='%(objectname)' 'refs/remotes/upstream/master' 'refs/remotes/upstream/*.x')"
+    mapfile -t branches <<<"$branches"
+    if merge_base="$(git merge-base -- HEAD "${branches[@]}")"; then
+        git describe --always --tags --match='[0-9]*' -- "$merge_base"
+    fi
+} >zulip-git-version
```

Comment:
What happens the first time you `upgrade-zulip-from-git`, where we don't have an `upstream` in `/srv/zulip.git` yet?

Reply:
`git merge-base` errors, the `if` condition is false, and you get a one-line `zulip-git-version`.

Reply:
OK.